h2o-r/h2o-package/R/infogram.R

```diff
@@ -0,0 +1,526 @@
+                         stopping_metric = c("AUTO", "deviance", "logloss", "MSE", "RMSE", "MAE", "RMSLE", "AUC", "AUCPR", "lift_top_group", "misclassification", "mean_per_class_error", "custom", "custom_increasing"),
+                         stopping_tolerance = 0.001,
+                         balance_classes = FALSE,
+                         class_sampling_factors = NULL,
+                         max_after_balance_size = 5.0,
+                         max_runtime_secs = 0,
+                         custom_metric_func = NULL,
+                         auc_type = c("AUTO", "NONE", "MACRO_OVR", "WEIGHTED_OVR", "MACRO_OVO", "WEIGHTED_OVO"),
+                         infogram_algorithm = c("AUTO", "deeplearning", "drf", "gbm", "glm", "xgboost"),
+                         infogram_algorithm_params = NULL,
+                         protected_columns = NULL,
+                         net_information_threshold = 4.940656458e-324,
+                         total_information_threshold = 4.940656458e-324,
+                         safety_index_threshold = 4.940656458e-324,
+                         relevance_index_threshold = 4.940656458e-324,
+                         data_fraction = 1,
+                         top_n_features = 50,
+                         compute_p_values = FALSE)
+{
+  # Validate required training_frame first and other frame args: should be a valid key or an H2OFrame object
+  training_frame <- .validate.H2OFrame(training_frame, required=TRUE)
+  validation_frame <- .validate.H2OFrame(validation_frame, required=FALSE)
+
+  # Validate other required args
+  # If x is missing, then assume user wants to use all columns as features.
+  if (missing(x)) {
+     if (is.numeric(y)) {
+         x <- setdiff(col(training_frame), y)
+     } else {
+         x <- setdiff(colnames(training_frame), y)
+     }
+  }
+
+  # Build parameter list to send to model builder
+  parms <- list()
+  parms$training_frame <- training_frame
+  args <- .verify_dataxy(training_frame, x, y)
+  if (missing(protected_columns)) { # core infogram
+    if (!missing(safety_index_threshold)) {
+      warning("Should not set safety_index_threshold for core infogram runs.  Set net_information_threshold instead.
+          Using default of 0.1 if not set")
+    }
+    if (!missing(relevance_index_threshold)) {
+      warning("Should not set relevance_index_threshold for core infogram runs.  Set total_information_threshold 
+      instead.   Using default of 0.1 if not set")
+    }
+  } else { # fair infogram
+    if (!missing(net_information_threshold)) {
+    warning("Should not set net_information_threshold for fair infogram runs, set safety_index_threshold instead.  
+      Using default of 0.1 if not set")
+    }
+    if (!missing(total_information_threshold)) {
+      warning("Should not set total_information_threshold for fair infogram runs, set relevance_index_threshold
+       instead.    Using default of 0.1 if not set")
```

Comment:
The warnings contain probably unwanted whitespaces leading to suboptimally formatted warnings, e.g.,
```
Warning message:
Should not set total_information_threshold for fair infogram runs, set relevance_index_threshold
        instead.    Using default of 0.1 if not set 
```
or 
```
Warning message:
Should not set net_information_threshold for fair infogram runs, set safety_index_threshold instead.  
       Using default of 0.1 if not set 
```

Reply:
Good catch.  Will fix.

Reply:
Fixed.